### PR TITLE
Sort out system tests + remaining test failures for pg

### PR DESCRIPTION
**Pull request by colinxfleming**
I rule and have completed some work on Case Manager that's ready for review!

*note: goes into the unit tests branch, not the shell branch.*

Couple more edits and we're home free! Or at least free in the sense that then we start really acid testing everything.

This pull request makes the following changes:
* systemtest updates
* fix a timezone problem in a test
* get brakeman to pass

It relates to the following issue #s: 
* Fixes #X
* Bumps #2072 

For reviewer:
* Adjust the title to explain what it does for the notification email to the listserv.
* Tag this PR:
  * `feature` if it contains a feature, fix, or similar. This is anything that contains a user-facing fix in some way, such as frontend changes, alterations to backend behavior, or bug fixes.
  * `dependencies` if it contains library upgrades or similar. This is anything that upgrades any dependency, such as a Gemfile update or npm package upgrade.
* If it contains neither, no need to tag this PR.

### CODE REVIEW

File: test/system/event_log_interaction_test.rb [hunk 1]
```diff
@@ -38,14 +37,42 @@ class EventLogInteractionTest < ApplicationSystemTestCase
...
-                              "#{@patient.name}"
-        end
         click_on '(Add to call list)'
         wait_for_ajax
         @user2.reload
       end
     end
   end
+
+  describe 'sending pledges' do
```
Comment (colinxfleming) [inline test/system/event_log_interaction_test.rb:47]: This is a net new test, I think.

File: test/system/event_log_interaction_test.rb [hunk 2]
```diff
@@ -38,14 +37,42 @@ class EventLogInteractionTest < ApplicationSystemTestCase
       end
 
       assert_difference '@user2.call_list_entries.count', 1 do
-        within :css, '#activity_log_content' do
```
Comment (colinxfleming) [inline test/system/event_log_interaction_test.rb:41]: Weird spurious failures here for a reason I couldn't figure out.

File: app/models/patient.rb [hunk 3]
```diff
@@ -49,10 +49,10 @@ class Patient < ApplicationRecord
             :initial_call_date,
             :line,
             presence: true
-  validates :primary_phone, format: /\d{10}/,
+  validates :primary_phone, format: /\A\d{10}\z/,
```
Comment (colinxfleming) [inline app/models/patient.rb:52]: Brakeman change - brakeman says 'keep it tight by anchoring start and end in your regexes'.

**Review by colinxfleming — COMMENTED** [review]
notes

File: test/system/event_log_interaction_test.rb (hunk 2 above)
Comment (lomky) [reply test/system/event_log_interaction_test.rb:41]: isnt this redundant to the test on line 35 above anyway?

**Review by lomky — APPROVED** [review]
:shipit:

**Pull request merged**